Fix warnings reported when compiling the test runner

There's a bunch of warnings reported by the compiler. It would be nice if we address them.

```
dotnet-bunny/Turkey/DotNet.cs(162,30): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' annotations context. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Cleaner.cs(39,27): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Cleaner.cs(56,27): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/NuGet.cs(28,33): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/NuGet.cs(38,35): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/ProcessExtensions.cs(50,47): warning CS0168: The variable 'ex' is declared but never used [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestOutputFormat.cs(81,40): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestOutputFormat.cs(94,40): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestOutputFormat.cs(16,40): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestOutputFormat.cs(22,40): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestOutputFormat.cs(51,40): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestRunner.cs(47,35): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestRunner.cs(48,35): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestRunner.cs(49,35): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestRunner.cs(50,35): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestRunner.cs(51,35): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestParser.cs(18,54): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread. [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestOutputFormat.cs(14,22): warning CA1034: Do not nest type NewOutput. Alternatively, change its accessibility so that it is not externally visible. (https://msdn.microsoft.com/en-us/library/ms182162.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestOutputFormat.cs(12,18): warning CA1052: Type 'TestOutputFormats' is a static holder type but is neither static nor NotInheritable (http://msdn.microsoft.com/library/ms182168.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestOutputFormat.cs(57,22): warning CA1034: Do not nest type JUnitOutput. Alternatively, change its accessibility so that it is not externally visible. (https://msdn.microsoft.com/en-us/library/ms182162.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Program.cs(15,18): warning CA1052: Type 'Program' is a static holder type but is neither static nor NotInheritable (http://msdn.microsoft.com/library/ms182168.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Test.cs(21,29): warning CA2227: Change 'IgnoredRIDs' to be read-only by removing the property setter. (https://msdn.microsoft.com/library/ms182327.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Test.cs(22,29): warning CA2227: Change 'SkipWhen' to be read-only by removing the property setter. (https://msdn.microsoft.com/library/ms182327.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestRunner.cs(25,52): warning CA1051: Do not declare visible instance fields (https://msdn.microsoft.com/en-us/library/ms182141.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/XUnitTest.cs(19,37): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/XUnitTest.cs(20,37): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/BashTest.cs(41,34): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/CsprojCompatibilityPatcher.cs(13,23): warning CA1822: Member Patch does not access instance data and can be marked as static (Shared in VisualBasic) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Cleaner.cs(39,27): warning CA1822: Member CleanProjectLocalDotNetCruftAsync does not access instance data and can be marked as static (Shared in VisualBasic) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Cleaner.cs(85,17): warning CA1307: The behavior of 'string.StartsWith(string)' could vary based on the current user's locale settings. Replace this call in 'Turkey.Cleaner.ExpandPath(string)' with a call to 'string.StartsWith(string, System.StringComparison)'. (https://msdn.microsoft.com/en-us/library/bb386080.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Cleaner.cs(83,36): warning CA1822: Member ExpandPath does not access instance data and can be marked as static (Shared in VisualBasic) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/DotNet.cs(46,40): warning CA1307: The behavior of 'string.StartsWith(string)' could vary based on the current user's locale settings. Replace this call in 'Turkey.DotNet.RuntimeVersions.get' with a call to 'string.StartsWith(string, System.StringComparison)'. (https://msdn.microsoft.com/en-us/library/bb386080.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/DotNet.cs(140,26): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/IEnumerableExtensions.cs(13,23): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/NuGet.cs(24,32): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/NuGet.cs(25,26): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/PlatformId.cs(21,27): warning CA1308: In method 'ComputePlatformIds', replace the call to 'ToLowerInvariant' with 'ToUpperInvariant'. (https://msdn.microsoft.com/en-us/library/bb386042.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/PlatformId.cs(66,24): warning CA1822: Member GetValue does not access instance data and can be marked as static (Shared in VisualBasic) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/PlatformId.cs(76,17): warning CA1307: The behavior of 'string.StartsWith(string)' could vary based on the current user's locale settings. Replace this call in 'Turkey.PlatformId.Unquote(string)' with a call to 'string.StartsWith(string, System.StringComparison)'. (https://msdn.microsoft.com/en-us/library/bb386080.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/PlatformId.cs(76,42): warning CA1307: The behavior of 'string.EndsWith(string)' could vary based on the current user's locale settings. Replace this call in 'Turkey.PlatformId.Unquote(string)' with a call to 'string.EndsWith(string, System.StringComparison)'. (https://msdn.microsoft.com/en-us/library/bb386080.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/PlatformId.cs(71,24): warning CA1822: Member Unquote does not access instance data and can be marked as static (Shared in VisualBasic) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/PlatformId.cs(84,25): warning CA1822: Member GetLddVersion does not access instance data and can be marked as static (Shared in VisualBasic) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/ProcessExtensions.cs(15,19): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/NuGet.cs(31,40): warning CA1307: The behavior of 'Newtonsoft.Json.Linq.JObject.GetValue(string?)' could vary based on the current user's locale settings. Replace this call in 'Turkey.NuGet.IsPackageLiveAsync(string, Turkey.Version, string)' with a call to 'Newtonsoft.Json.Linq.JObject.GetValue(string?, System.StringComparison)'. (https://msdn.microsoft.com/en-us/library/bb386080.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/NuGet.cs(33,29): warning CA1307: The behavior of 'string.Equals(string?)' could vary based on the current user's locale settings. Replace this call in 'Turkey.NuGet.IsPackageLiveAsync(string, Turkey.Version, string)' with a call to 'string.Equals(string?, System.StringComparison)'. (https://msdn.microsoft.com/en-us/library/bb386080.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/NuGet.cs(28,33): warning CA1822: Member IsPackageLiveAsync does not access instance data and can be marked as static (Shared in VisualBasic) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/NuGet.cs(24,32): warning CA2234: Modify 'NuGet.IsPackageLiveAsync(string, Version)' to call 'HttpClient.GetStringAsync(Uri)' instead of 'HttpClient.GetStringAsync(string)'. (https://msdn.microsoft.com/en-us/library/ms182360.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/SourceBuild.cs(19,23): warning CA1055: Change the return type of method SourceBuild.GetBranchContentUrl(Version) from string to System.Uri. (https://msdn.microsoft.com/en-us/library/ms182176.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/SourceBuild.cs(19,23): warning CA1822: Member GetBranchContentUrl does not access instance data and can be marked as static (Shared in VisualBasic) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Version.cs(137,13): warning CA1065: GetHashCode creates an exception of type Exception. Exceptions should not be raised in this type of method. If this exception instance might be raised, change this method's logic so it no longer raises an exception. (https://msdn.microsoft.com/en-us/library/bb386039.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/ProcessExtensions.cs(46,23): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/SourceBuild.cs(44,33): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/SourceBuild.cs(46,40): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/SourceBuild.cs(63,37): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/NuGet.cs(41,23): warning CA2208: Call the ArgumentNullException constructor that contains a message and/or paramName parameter. (https://msdn.microsoft.com/en-us/library/ms182347.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/NuGet.cs(69,31): warning CA1307: The behavior of 'string.Replace(string, string?)' could vary based on the current user's locale settings. Replace this call in 'Turkey.NuGet.GenerateNuGetConfig(System.Collections.Generic.List<string>, string)' with a call to 'string.Replace(string, string?, System.StringComparison)'. (https://msdn.microsoft.com/en-us/library/bb386080.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/SourceBuild.cs(63,37): warning CA2234: Modify 'SourceBuild.GetNuGetConfigAsync(Version)' to call 'HttpClient.GetStringAsync(Uri)' instead of 'HttpClient.GetStringAsync(string)'. (https://msdn.microsoft.com/en-us/library/ms182360.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/NuGet.cs(38,35): warning CA1822: Member GenerateNuGetConfig does not access instance data and can be marked as static (Shared in VisualBasic) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/SourceBuild.cs(44,33): warning CA2234: Modify 'SourceBuild.GetProdConFeedAsync(Version)' to call 'HttpClient.GetStringAsync(Uri)' instead of 'HttpClient.GetStringAsync(string)'. (https://msdn.microsoft.com/en-us/library/ms182360.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/SourceBuild.cs(46,40): warning CA2234: Modify 'SourceBuild.GetProdConFeedAsync(Version)' to call 'HttpClient.GetAsync(Uri)' instead of 'HttpClient.GetAsync(string)'. (https://msdn.microsoft.com/en-us/library/ms182360.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Program.cs(117,40): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Program.cs(132,33): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Test.cs(61,23): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Test.cs(66,36): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestOutputFormat.cs(149,28): warning CA1822: Member RemoveInvalidXmlCharacters does not access instance data and can be marked as static (Shared in VisualBasic) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Program.cs(156,48): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Program.cs(176,45): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Program.cs(192,34): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestOutputFormat.cs(105,58): warning CA1305: The behavior of 'int.ToString()' could vary based on the current user's locale settings. Replace this call in 'TestOutputFormats.JUnitOutput.AfterRunningAllTestsAsync(TestResults)' with a call to 'int.ToString(IFormatProvider)'. (https://msdn.microsoft.com/en-us/library/ms182190.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestOutputFormat.cs(106,61): warning CA1305: The behavior of 'int.ToString()' could vary based on the current user's locale settings. Replace this call in 'TestOutputFormats.JUnitOutput.AfterRunningAllTestsAsync(TestResults)' with a call to 'int.ToString(IFormatProvider)'. (https://msdn.microsoft.com/en-us/library/ms182190.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestOutputFormat.cs(18,32): warning CA1305: The behavior of 'string.Format(string, object)' could vary based on the current user's locale settings. Replace this call in 'TestOutputFormats.NewOutput.AfterParsingTestAsync(string, bool)' with a call to 'string.Format(IFormatProvider, string, params object[])'. (https://msdn.microsoft.com/en-us/library/ms182190.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestRunner.cs(73,19): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Program.cs(218,27): warning CA1308: In method 'CreateTraits', replace the call to 'ToLowerInvariant' with 'ToUpperInvariant'. (https://msdn.microsoft.com/en-us/library/bb386042.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestRunner.cs(94,23): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestRunner.cs(97,40): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestRunner.cs(106,23): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestRunner.cs(126,27): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestRunner.cs(132,40): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestRunner.cs(150,23): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestRunner.cs(153,19): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Program.cs(235,31): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Program.cs(238,30): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestParser.cs(101,22): warning CA1822: Member VersionMatches does not access instance data and can be marked as static (Shared in VisualBasic) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Program.cs(261,26): warning CA2007: Do not directly await a Task without calling ConfigureAwait [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/TestParser.cs(18,144): warning CA1801: Parameter testConfiguration of method TryParseAsync is never used. Remove the parameter or use it in the method body. (https://msdn.microsoft.com/en-us/library/ms182268.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/Program.cs(199,89): warning CA1801: Parameter sdkVersion of method CreateTraits is never used. Remove the parameter or use it in the method body. (https://msdn.microsoft.com/en-us/library/ms182268.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
dotnet-bunny/Turkey/NuGet.cs(28,59): warning CA1801: Parameter name of method IsPackageLiveAsync is never used. Remove the parameter or use it in the method body. (https://msdn.microsoft.com/en-us/library/ms182268.aspx) [dotnet-bunny/Turkey/Turkey.csproj]
```

cc @omajid

--- a/Turkey/BashTest.cs
+++ b/Turkey/BashTest.cs
@@ -38,7 +38,7 @@ namespace Turkey
                 startInfo.EnvironmentVariables.Add(key, value);
             }
 
-            int exitCode = await ProcessRunner.RunAsync(startInfo, logger, cancellationToken);
+            int exitCode = await ProcessRunner.RunAsync(startInfo, logger, cancellationToken).ConfigureAwait(false);
 
             return exitCode == 0 ? TestResult.Passed : TestResult.Failed;
         }

--- a/Turkey/Cleaner.cs
+++ b/Turkey/Cleaner.cs
@@ -36,7 +36,9 @@ namespace Turkey
             yield return "project.lock.json";
         }
 
-        public async Task CleanProjectLocalDotNetCruftAsync()
+#pragma warning disable CA1822 // Mark members as static
+        public Task CleanProjectLocalDotNetCruftAsync()
+#pragma warning restore CA1822 // Mark members as static
         {
 
             foreach(var name in LocalProjectCruft())
@@ -51,9 +53,12 @@ namespace Turkey
                     File.Delete(name);
                 }
             }
+            return Task.CompletedTask;
         }
 
-        public async Task CleanLocalDotNetCacheAsync()
+#pragma warning disable CA1822 // Mark members as static
+        public Task CleanLocalDotNetCacheAsync()
+#pragma warning restore CA1822 // Mark members as static
         {
             foreach (var path in CruftDirectoryGlobs())
             {
@@ -77,12 +82,14 @@ namespace Turkey
                     Console.WriteLine($"WARNING: unable to expand {path}");
                 }
             }
-            return;
+            return Task.CompletedTask;
         }
 
+#pragma warning disable CA1822 // Mark members as static
         public IEnumerable<string> ExpandPath(string pathWithGlob)
+#pragma warning restore CA1822 // Mark members as static
         {
-            if (pathWithGlob.StartsWith("~"))
+            if (pathWithGlob.StartsWith("~", StringComparison.Ordinal))
             {
                 pathWithGlob = Environment.GetEnvironmentVariable("HOME") + pathWithGlob.Substring(1);
             }

--- a/Turkey/DotNet.cs
+++ b/Turkey/DotNet.cs
@@ -43,7 +43,7 @@ namespace Turkey
                     string output = p.StandardOutput.ReadToEnd();
                     var list = output
                         .Split("\n", StringSplitOptions.RemoveEmptyEntries)
-                        .Where(line => line.StartsWith("Microsoft.NETCore.App"))
+                        .Where(line => line.StartsWith("Microsoft.NETCore.App", StringComparison.Ordinal))
                         .Select(line => line.Split(" ")[1])
                         .Select(versionString => Version.Parse(versionString))
                         .OrderBy(x => x)
@@ -137,7 +137,7 @@ namespace Turkey
                 startInfo.EnvironmentVariables.Add(key, value);
             }
 
-            return await ProcessRunner.RunAsync(startInfo, logger, token);
+            return await ProcessRunner.RunAsync(startInfo, logger, token).ConfigureAwait(false);
         }
 
         private static bool IsCoreClrRuntime(string dotnetRoot, Version version)
@@ -159,7 +159,9 @@ namespace Turkey
             return File.Exists(Path.Combine(runtimeDir, "libcoreclrtraceptprovider.so"));
         }
 
+        #nullable enable
         private static string? FindProgramInPath(string program)
+        #nullable disable
         {
             string[] paths = Environment.GetEnvironmentVariable("PATH")?.Split(':', StringSplitOptions.RemoveEmptyEntries) ?? Array.Empty<string>();
             foreach (string p in paths)

--- a/Turkey/IEnumerableExtensions.cs
+++ b/Turkey/IEnumerableExtensions.cs
@@ -10,7 +10,7 @@ namespace Turkey
         {
             foreach (T item in items)
             {
-                await task(item);
+                await task(item).ConfigureAwait(false);
             }
         }
     }

--- a/Turkey/NuGet.cs
+++ b/Turkey/NuGet.cs
@@ -21,27 +21,34 @@ namespace Turkey
         public async Task<bool> IsPackageLiveAsync(string name, Version version)
         {
             var url = $"https://api-v2v3search-0.nuget.org/autocomplete?id={name}&prerelease=true";
-            var result = await _client.GetStringAsync(url);
-            return await IsPackageLiveAsync(name, version, result);
+            Uri uri = new(url);
+            var result = await _client.GetStringAsync(uri).ConfigureAwait(false);
+            return await IsPackageLiveAsync(name, version, result).ConfigureAwait(false);
         }
 
-        public async Task<bool> IsPackageLiveAsync(string name, Version version, string json)
+#pragma warning disable CA1801 // Remove unused parameter
+#pragma warning disable CA1822 // Mark members as static
+        public Task<bool> IsPackageLiveAsync(string name, Version version, string json)
+#pragma warning restore CA1822 // Mark members as static
+#pragma warning restore CA1801 // Remove unused parameter
         {
             JObject deserialized = (JObject) JsonConvert.DeserializeObject(json);
-            JArray versions = (JArray) deserialized.GetValue("data");
+            JArray versions = (JArray) deserialized.GetValue("data", StringComparison.Ordinal);
             var found = versions.Children<JToken>()
-                .Where(v => v.Value<string>().Equals(version.ToString()))
+                .Where(v => v.Value<string>().Equals(version.ToString(), StringComparison.Ordinal))
                 .Any();
-            return found;
+            return Task.FromResult(found);
         }
 
-        public async Task<string> GenerateNuGetConfig(List<string> urls, string nugetConfig = null)
+#pragma warning disable CA1822 // Mark members as static
+        public Task<string> GenerateNuGetConfig(List<string> urls, string nugetConfig = null)
+#pragma warning restore CA1822 // Mark members as static
         {
-            if( !urls.Any() && nugetConfig == null )
-                throw new ArgumentNullException();
+            if (!urls.Any())
+                ArgumentNullException.ThrowIfNull(nugetConfig);
 
             string sources = null;
-            if( urls.Any() )
+            if (urls.Any())
             {
                 var sourceParts = new List<string>(urls.Count);
                 for( int i = 0; i < urls.Count; i++ )
@@ -54,6 +61,7 @@ namespace Turkey
                 {
                     sources = $"    {sources}\n";
                 }
+            
             }
 
             if( string.IsNullOrWhiteSpace(nugetConfig) )
@@ -66,9 +74,9 @@ namespace Turkey
             }
 
             if( !string.IsNullOrWhiteSpace(sources) )
-                nugetConfig = nugetConfig.Replace("</packageSources>", sources + "</packageSources>");
+                nugetConfig = nugetConfig.Replace("</packageSources>", sources + "</packageSources>", StringComparison.Ordinal);
 
-            return nugetConfig;
+            return Task.FromResult(nugetConfig);
         }
 
     }

--- a/Turkey/PlatformId.cs
+++ b/Turkey/PlatformId.cs
@@ -18,7 +18,9 @@ namespace Turkey
 
         public List<string> ComputePlatformIds(string[] osReleaseLines, string lddVersionOutput)
         {
-            string arch = Enum.GetName(typeof(Architecture), RuntimeInformation.OSArchitecture).ToLowerInvariant();
+            #pragma warning disable CA1308 // Normalize strings to uppercase
+            string arch = RuntimeInformation.OSArchitecture.ToString().ToLowerInvariant();
+            #pragma warning restore CA1308 // Normalize strings to uppercase
             return ComputePlatformIds(osReleaseLines, arch, lddVersionOutput);
         }
 
@@ -63,17 +65,17 @@ namespace Turkey
             return platforms.ToList();
         }
 
-        private string GetValue(string key, string[] lines)
+        private static string GetValue(string key, string[] lines)
         {
             return lines.Where(line => line.StartsWith(key + "=", StringComparison.Ordinal)).Last().Substring((key + "=").Length);
         }
 
-        private string Unquote(string text)
+        private static string Unquote(string text)
         {
             // TODO implement proper un-escaping
             // This is a limited shell-style syntax described at
             // https://www.freedesktop.org/software/systemd/man/os-release.html
-            if (text.StartsWith("\"") && text.EndsWith("\""))
+            if (text.StartsWith("\"", StringComparison.Ordinal) && text.EndsWith("\"", StringComparison.Ordinal))
             {
                 return text.Substring(1, text.Length - 2);
             }
@@ -81,7 +83,9 @@ namespace Turkey
             return text;
         }
 
+#pragma warning disable CA1822 // Mark members as static
         internal string GetLddVersion()
+#pragma warning restore CA1822 // Mark members as static
         {
             using (Process p = new Process())
             {

--- a/Turkey/ProcessExtensions.cs
+++ b/Turkey/ProcessExtensions.cs
@@ -12,7 +12,7 @@ namespace Turkey
         {
             logger($"Executing {psi.FileName} with arguments {psi.Arguments} in working directory {psi.WorkingDirectory}");
             using var process = Process.Start(psi);
-            await process.WaitForExitAsync(logger, token);
+            await process.WaitForExitAsync(logger, token).ConfigureAwait(false);
             return process.ExitCode;
         }
     }
@@ -43,11 +43,11 @@ namespace Turkey
 
             try
             {
-                await process.WaitForExitAsync(token);
+                await process.WaitForExitAsync(token).ConfigureAwait(false);
 
                 logger($"Process Exit Code: {process.ExitCode}");
             }
-            catch (OperationCanceledException ex)
+            catch (OperationCanceledException)
             {
                 lock (logger)
                 {

--- a/Turkey/Program.cs
+++ b/Turkey/Program.cs
@@ -12,7 +12,9 @@ using System.Runtime.InteropServices;
 
 namespace Turkey
 {
+#pragma warning disable CA1052 // Static holder types should be Static or NotInheritable
     public class Program
+#pragma warning restore CA1052 // Static holder types should be Static or NotInheritable
     {
         public static readonly Option<bool> verboseOption = new Option<bool>(
             new string[] { "--verbose", "-v" },
@@ -118,7 +120,7 @@ namespace Turkey
 
             Version packageVersion = runtimeVersion;
             string nuGetConfig = await GenerateNuGetConfigIfNeededAsync(additionalFeed, packageVersion,
-                                                                        useSourceBuildNuGetConfig: false);
+                                                                        useSourceBuildNuGetConfig: false).ConfigureAwait(false);
             if (verbose && nuGetConfig != null)
             {
                 Console.WriteLine("Using nuget.config: ");
@@ -132,7 +134,7 @@ namespace Turkey
                 verboseOutput: verbose,
                 nuGetConfig: nuGetConfig);
 
-            var results = await runner.ScanAndRunAsync(testOutputs, logDir.FullName, defaultTimeout);
+            var results = await runner.ScanAndRunAsync(testOutputs, logDir.FullName, defaultTimeout).ConfigureAwait(false);
 
             int exitCode = (results.Failed == 0) ? 0 : 1;
             return exitCode;
@@ -157,7 +159,7 @@ namespace Turkey
                 {
                     try
                     {
-                        nugetConfig = await sourceBuild.GetNuGetConfigAsync(netCoreAppVersion);
+                        nugetConfig = await sourceBuild.GetNuGetConfigAsync(netCoreAppVersion).ConfigureAwait(false);
                     }
                     catch( HttpRequestException exception )
                     {
@@ -173,14 +175,16 @@ namespace Turkey
                     // if the nugetConfig has a <clear/> element that removes
                     // it.
                     urls.Add("https://api.nuget.org/v3/index.json");
-                    return await nuget.GenerateNuGetConfig(urls, nugetConfig);
+                    return await nuget.GenerateNuGetConfig(urls, nugetConfig).ConfigureAwait(false);
                 }
             }
 
             return null;
         }
 
+#pragma warning disable CA1801 // Remove unused parameter
         public static IReadOnlySet<string> CreateTraits(Version runtimeVersion, Version sdkVersion, List<string> rids, bool isMonoRuntime, IEnumerable<string> additionalTraits)
+#pragma warning restore CA1801 // Remove unused parameter
         {
             var traits = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
 
@@ -199,7 +203,9 @@ namespace Turkey
             }
 
             // Add 'arch=' trait.
+#pragma warning disable CA1308 // Normalize strings to uppercase
             string arch = RuntimeInformation.OSArchitecture.ToString().ToLowerInvariant();
+#pragma warning restore CA1308 // Normalize strings to uppercase
             traits.Add($"arch={arch}");
 
             // Add 'runtime=' trait.
@@ -231,7 +237,7 @@ namespace Turkey
             rootCommand.AddOption(traitOption);
             rootCommand.AddOption(timeoutOption);
 
-            return await rootCommand.InvokeAsync(args);
+            return await rootCommand.InvokeAsync(args).ConfigureAwait(false);
         }
     }
 }

--- a/Turkey/SourceBuild.cs
+++ b/Turkey/SourceBuild.cs
@@ -16,11 +16,12 @@ namespace Turkey
             this._client = client;
         }
 
-        public string GetBranchContentUrl(Version version)
+        public static System.Uri GetBranchContentUrl(Version version)
         {
             var branchName = "release/" + version.MajorMinor + ".1xx";
             var url = $"https://raw.githubusercontent.com/dotnet/installer/{branchName}/";
-            return url;
+            Uri uri = new(url);
+            return uri;
         }
 
         public async Task<string> GetProdConFeedAsync(Version version)
@@ -31,9 +32,10 @@ namespace Turkey
             }
 
             var url = GetBranchContentUrl(version) + "ProdConFeed.txt";
-            var feedUrl = await _client.GetStringAsync(url);
-
-            using(var response = await _client.GetAsync(feedUrl))
+            Uri uri = new(url);
+            var feedUrl = await _client.GetStringAsync(uri).ConfigureAwait(false);
+            Uri feedUri = new(feedUrl);
+            using(var response = await _client.GetAsync(feedUri).ConfigureAwait(false))
             {
                 if (!response.IsSuccessStatusCode)
                 {
@@ -46,11 +48,12 @@ namespace Turkey
         public async Task<string> GetNuGetConfigAsync(Version version)
         {
             string url = GetBranchContentUrl(version) + "NuGet.config";
+            Uri uri = new(url);
 
             string nugetConfig = null;
             try
             {
-                nugetConfig = await _client.GetStringAsync(url);
+                nugetConfig = await _client.GetStringAsync(uri).ConfigureAwait(false);
             }
             catch( HttpRequestException e )
             {

--- a/Turkey/Test.cs
+++ b/Turkey/Test.cs
@@ -18,8 +18,12 @@ namespace Turkey
         public string Type { get; set; }
         public bool Cleanup { get; set; }
         public double TimeoutMultiplier { get; set; } = 1.0;
+
+        #pragma warning disable CA2227 // Change to be read-only by removing the property setter.
         public List<string> IgnoredRIDs { get; set; } = new();
         public List<string> SkipWhen { get; set; } = new();
+
+        #pragma warning restore CA2227
     }
 
     // TODO is this a strongly-typed enum in C#?
@@ -58,10 +62,10 @@ namespace Turkey
                 {
                     Console.WriteLine($"WARNING: overwriting {path}");
                 }
-                await File.WriteAllTextAsync(path, NuGetConfig);
+                await File.WriteAllTextAsync(path, NuGetConfig).ConfigureAwait(false);
             }
 
-            var testResult = await InternalRunAsync(logger, cancelltionToken);
+            var testResult = await InternalRunAsync(logger, cancelltionToken).ConfigureAwait(false);
 
             if (!string.IsNullOrEmpty(NuGetConfig))
             {

--- a/Turkey/TestOutputFormat.cs
+++ b/Turkey/TestOutputFormat.cs
@@ -11,9 +11,9 @@ using System.Xml.Serialization;
 
 namespace Turkey
 {
-    public class TestOutputFormats
+    public static class TestOutputFormats
     {
-        public class NewOutput : TestOutput
+        internal class NewOutput : TestOutput
         {
             public class FailedTest
             {
@@ -23,16 +23,23 @@ namespace Turkey
 
             private List<FailedTest> failedTests = new List<FailedTest>();
 
-            public async override Task AtStartupAsync(){
-                Console.WriteLine("Running tests:");
-            }
-            public async override Task AfterParsingTestAsync(string name, bool enabled)
+#pragma warning disable CA1822 // Mark members as static
+            public void AtStartup()
+#pragma warning restore CA1822 // Mark members as static
             {
+                Console.WriteLine("Running tests:");
+                return;
+            }
+            public override Task AfterParsingTestAsync(string name, bool enabled)
+            {
+                #pragma warning disable CA1305
                 var nameText = string.Format("{0,-60}", name);
+                #pragma warning restore CA1305
                 Console.Write(nameText);
+                return Task.CompletedTask;
             }
 
-            public async override Task AfterRunningTestAsync(string name, TestResult result, StringBuilder testLog, TimeSpan testTime)
+            public override Task AfterRunningTestAsync(string name, TestResult result, StringBuilder testLog, TimeSpan testTime)
             {
                 int minutes = (int)testTime.TotalMinutes;
                 int seconds = (int)Math.Ceiling(testTime.TotalSeconds - 60 * minutes);
@@ -59,26 +66,31 @@ namespace Turkey
                     }
                     Console.WriteLine($"[{resultOutput}]\t({elapsedTime})");
                 }
+                return Task.CompletedTask;
             }
 
-            public async override Task PrintFailedTests()
+            public override Task PrintFailedTests()
             {
                 Console.WriteLine();
                 Console.WriteLine("The following tests failed: ");
                 foreach(var test in failedTests)
                 {
+                    #pragma warning disable CA1305
                     Console.WriteLine($"{string.Format("{0,-30}", test.Name)}({test.Duration})");
+                    #pragma warning restore CA1305
                 }
+                return Task.CompletedTask;
             }
 
-            public async override Task AfterRunningAllTestsAsync(TestResults results)
+            public override Task AfterRunningAllTestsAsync(TestResults results)
             {
                 Console.WriteLine();
                 Console.WriteLine($"Total: {results.Total} Passed: {results.Passed} Failed: {results.Failed}");
+                return Task.CompletedTask;
             }
         }
 
-        public class JUnitOutput : TestOutput
+        internal class JUnitOutput : TestOutput
         {
             private struct TestCase {
                 public string Name;
@@ -102,7 +114,7 @@ namespace Turkey
                 _resultsFile = resultsFile;
             }
 
-            public async override Task AfterRunningTestAsync(string name, TestResult result, StringBuilder testLog, TimeSpan testTime)
+            public override Task AfterRunningTestAsync(string name, TestResult result, StringBuilder testLog, TimeSpan testTime)
             {
                 var testCase = new TestCase();
                 testCase.Name = name;
@@ -113,9 +125,11 @@ namespace Turkey
                 testCase.Log = testLog;
 
                 _testCases.Add(testCase);
+
+                return Task.CompletedTask;
             }
 
-            public async override Task AfterRunningAllTestsAsync(TestResults results)
+            public override Task AfterRunningAllTestsAsync(TestResults results)
             {
                 var settings = new XmlWriterSettings();
                 settings.Indent = true;
@@ -126,8 +140,10 @@ namespace Turkey
 
                     writer.WriteStartElement("testsuite");
                     writer.WriteAttributeString("name", "dotnet");
+#pragma warning disable CA1305 // Specify IFormatProvider
                     writer.WriteAttributeString("tests", _testCases.Count.ToString());
                     writer.WriteAttributeString("failures", _testCases.Where(t => t.Failed).Count().ToString());
+#pragma warning restore CA1305 // Specify IFormatProvider
                     writer.WriteAttributeString("errors", "0");
 
                     foreach (var testCase in _testCases)
@@ -168,9 +184,10 @@ namespace Turkey
                     writer.WriteEndDocument();
                     writer.Close();
                 }
+                return Task.CompletedTask;
             }
 
-            private string RemoveInvalidXmlCharacters(string input)
+            private static string RemoveInvalidXmlCharacters(string input)
             {
                 return Regex.Replace(input, @"[\u0000-\u0008,\u000B,\u000C,\u000E-\u001F]", "");
             }

--- a/Turkey/TestParser.cs
+++ b/Turkey/TestParser.cs
@@ -1,6 +1,7 @@
 using System;
 using System.IO;
 using System.Linq;
+using System.Reflection.Metadata.Ecma335;
 using System.Threading.Tasks;
 
 using Newtonsoft.Json;
@@ -15,7 +16,9 @@ namespace Turkey
             return TryParseAsync(system, nuGetConfig, dir, File.ReadAllText(testConfiguration.FullName));
         }
 
-        public async Task<(bool Success, Test Test)> TryParseAsync(SystemUnderTest system, string nuGetConfig, DirectoryInfo directory, string testConfiguration)
+#pragma warning disable CA1801 // Remove unused parameter
+        public Task<(bool Success, Test Test)> TryParseAsync(SystemUnderTest system, string nuGetConfig, DirectoryInfo directory, string testConfiguration)
+#pragma warning restore CA1801 // Remove unused parameter
         {
             // TODO: async
             var fileName = Path.Combine(directory.FullName, "test.json");
@@ -32,12 +35,12 @@ namespace Turkey
             {
                 case "xunit":
                     test = new XUnitTest(directory, system, nuGetConfig, descriptor, enabled);
-                    return (true, test);
+                    return Task.FromResult((true, test));
                 case "bash":
                     test = new BashTest(directory, system, nuGetConfig, descriptor, enabled);
-                    return (true, test);
+                    return Task.FromResult((true, test));
                 default:
-                    return (false, null);
+                    return Task.FromResult((false, (Test)null));
             }
         }
 
@@ -98,7 +101,9 @@ namespace Turkey
         }
 
 
+#pragma warning disable CA1822 // Mark members as static
         private bool VersionMatches(TestDescriptor test, Version runtimeVersion)
+#pragma warning restore CA1822 // Mark members as static
         {
             if (test.VersionSpecific)
             {

--- a/Turkey/TestRunner.cs
+++ b/Turkey/TestRunner.cs
@@ -22,7 +22,7 @@ namespace Turkey
         public Version RuntimeVersion { get; }
         public Version SdkVersion { get; }
         public List<string> CurrentPlatformIds { get; }
-        public IReadOnlyDictionary<string, string> EnvironmentVariables;
+        internal IReadOnlyDictionary<string, string> EnvironmentVariables;
         public IReadOnlySet<string> Traits { get; }
         public DotNet Dotnet { get; }
 
@@ -44,12 +44,12 @@ namespace Turkey
 
     public class TestOutput
     {
-        public async virtual Task AtStartupAsync() {}
-        public async virtual Task BeforeTestAsync() {}
-        public async virtual Task AfterParsingTestAsync(string name, bool enabled) {}
-        public async virtual Task AfterRunningTestAsync(string name, TestResult result, StringBuilder testLog, TimeSpan testTime) {}
-        public async virtual Task PrintFailedTests() {}
-        public async virtual Task AfterRunningAllTestsAsync(TestResults results) {}
+        public virtual Task AtStartupAsync() { return Task.CompletedTask; }
+        public virtual Task BeforeTestAsync() { return Task.CompletedTask; }
+        public virtual Task AfterParsingTestAsync(string name, bool enabled) { return Task.CompletedTask; }
+        public virtual Task AfterRunningTestAsync(string name, TestResult result, StringBuilder testLog, TimeSpan testTime) { return Task.CompletedTask; }
+        public virtual Task PrintFailedTests() { return Task.CompletedTask; }
+        public virtual Task AfterRunningAllTestsAsync(TestResults results) { return Task.CompletedTask; }
     }
 
     public class TestRunner
@@ -72,7 +72,7 @@ namespace Turkey
         public async Task<TestResults> ScanAndRunAsync(List<TestOutput> outputs, string logDir, TimeSpan defaultTimeout)
         {
 
-            await outputs.ForEachAsync(output => output.AtStartupAsync());
+            await outputs.ForEachAsync(output => output.AtStartupAsync()).ConfigureAwait(false);
 
             TestResults results = new TestResults();
 
@@ -93,10 +93,10 @@ namespace Turkey
             foreach (var file in sortedFiles)
             {
                 testTimeWatch.Reset();
-                await cleaner.CleanLocalDotNetCacheAsync();
+                await cleaner.CleanLocalDotNetCacheAsync().ConfigureAwait(false);
 
                 testTimeWatch.Start();
-                var parsedTest = await parser.TryParseAsync(system, nuGetConfig, file);
+                var parsedTest = await parser.TryParseAsync(system, nuGetConfig, file).ConfigureAwait(false);
                 if (!parsedTest.Success)
                 {
                     Console.WriteLine($"WARNING: Unable to parse {file}");
@@ -105,7 +105,7 @@ namespace Turkey
                 var test = parsedTest.Test;
                 string testName = test.Descriptor.Name;
 
-                await outputs.ForEachAsync(output => output.AfterParsingTestAsync(testName, !test.Skip));
+                await outputs.ForEachAsync(output => output.AfterParsingTestAsync(testName, !test.Skip)).ConfigureAwait(false);
 
                 TimeSpan testTimeout = test.Descriptor.TimeoutMultiplier * defaultTimeout;
                 using var cts = testTimeout == TimeSpan.Zero ? null : new CancellationTokenSource(testTimeout);
@@ -125,13 +125,13 @@ namespace Turkey
 
                 if (test.Descriptor.Cleanup)
                 {
-                    await cleaner.CleanProjectLocalDotNetCruftAsync();
+                    await cleaner.CleanProjectLocalDotNetCruftAsync().ConfigureAwait(false);
                 }
 
                 TestResult testResult;
                 try
                 {
-                    testResult = await test.RunAsync(testLogger, cancellationToken);
+                    testResult = await test.RunAsync(testLogger, cancellationToken).ConfigureAwait(false);
                 }
                 catch (OperationCanceledException)
                 {
@@ -149,15 +149,15 @@ namespace Turkey
                     case TestResult.Skipped: results.Skipped++; break;
                 }
 
-                await outputs.ForEachAsync(output => output.AfterRunningTestAsync(testName, testResult, testLog, testTimeWatch.Elapsed));
+                await outputs.ForEachAsync(output => output.AfterRunningTestAsync(testName, testResult, testLog, testTimeWatch.Elapsed)).ConfigureAwait(false);
                 }
             
             if (results.Failed != 0 )
             {
-                await outputs.ForEachAsync(outputs => outputs.PrintFailedTests());
+                await outputs.ForEachAsync(outputs => outputs.PrintFailedTests()).ConfigureAwait(false);
             }
 
-            await outputs.ForEachAsync(output => output.AfterRunningAllTestsAsync(results));
+            await outputs.ForEachAsync(output => output.AfterRunningAllTestsAsync(results)).ConfigureAwait(false);
 
             return results;
         }

--- a/Turkey/Version.cs
+++ b/Turkey/Version.cs
@@ -134,7 +134,9 @@ namespace Turkey
 
         public override int GetHashCode()
         {
+#pragma warning disable CA1065 // Do not raise exceptions in unexpected locations
             throw new NotImplementedException();
+#pragma warning restore CA1065 // Do not raise exceptions in unexpected locations
         }
 
     }

--- a/Turkey/XUnitTest.cs
+++ b/Turkey/XUnitTest.cs
@@ -16,8 +16,8 @@ namespace Turkey
 
         protected override async Task<TestResult> InternalRunAsync(Action<string> logger, CancellationToken cancellationToken)
         {
-            bool success =    await BuildProjectAsync(logger, cancellationToken) == 0
-                           && await RunTestProjectAsync(logger, cancellationToken) == 0;
+            bool success =    await BuildProjectAsync(logger, cancellationToken).ConfigureAwait(false) == 0
+                           && await RunTestProjectAsync(logger, cancellationToken).ConfigureAwait(false) == 0;
 
             return success ? TestResult.Passed : TestResult.Failed;
         }

